File: customised/velvet_app/flutter_app/lib/models/notification_template.dart
class NotificationTemplate {
  final String id;
  final String name;
  final String title;
  final String body;

  const NotificationTemplate({
    required this.id,
    required this.name,
    required this.title,
    required this.body,
  });

  factory NotificationTemplate.fromJson(Map<String, dynamic> json) {
    return NotificationTemplate(
      id: json['id'] as String? ?? '',
      name: json['name'] as String? ?? '',
      title: json['title'] as String? ?? '',
      body: json['body'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() => {
    'name': name,
    'title': title,
    'body': body,
  };
}

File: customised/velvet_app/dotnet_api/Models/NotificationTemplate.cs
namespace dotnet_api.Models;

public class NotificationTemplate
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string ClientId { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Body { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Client Client { get; set; } = null!;
}
